### ISSUE TITLE
nginx doesn't start

### ISSUE THREAD

**Issue by zakabluk** [bunkerity/bunkerized-nginx#4]
Whe I launched nginx with parameters:

```
        "SERVE_FILES" = "no"
        "DISABLE_DEFAULT_SERVER" = "yes"
        "USE_AUTH_BASIC" = "yes"
        "SERVER_NAME" = "elk-nomad-pro elk-nomad"
        "DISABLE_DEFAULT_SERVER" = "yes"
        "AUTH_BASIC_USER" = "nomad"
        "AUTH_BASIC_PASSWORD" = "mimimi"
        "AUTH_BASIC_TEXT" = "This is a nomad WebUI. elk-nomad-pro - is default UI, elk-nomad - is modern "
        "USE_MODSECURITY" = "yes"
```
and proxypass config

```
location / {
	if ($host = elk-nomad-pro) {
      proxy_pass http://nomad-ws$request_uri;
	}

	if ($host = elk-nomad) {
	  proxy_pass http://10.222.100.1:3000$request_uri;
	}

      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_read_timeout 310s;
      proxy_buffering off;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_set_header Origin "${scheme}://${proxy_host}";

}
```
after start container I get error message

`nginx: [emerg] duplicate location "/" in /etc/nginx/auth-basic.conf:1`

```
/etc/nginx # cat /etc/nginx/auth-basic.conf
location / {
	auth_basic "This is a nomad WebUI. elk-nomad-pro - is default UI, elk-nomad - is modern ";
	auth_basic_user_file /etc/nginx/.htpasswd;
}
```
my goal is it secur access to webui by password.

**Comment by bunkerity** [comment]
Hello @zakabluk can you try the latest release with the `AUTH_BASIC_LOCATION` environment variable set to *sitewide* (should be the default) ?

**Comment by zakabluk** [comment]
Hi! I made some tests.
1. Added AUTH_BASIC_LOCATION = /
`Error nginx: [emerg] duplicate location "/" in /etc/nginx/auth-basic.conf:1. `
 is remain.
2. Added AUTH_BASIC_LOCATION = ~* /(ui)
`Sep 30 21:56:26 nginx: 2020/09/30 21:56:26 [error] 1004#0: *18 "/etc/nginx/html/ui/index.html" is not found (2: No such file or directory), client: XXXX, server: elk-nomad-pro, request: "GET /ui/ HTTP/2.0", host: "elk-nomad"`
3 Added AUTH_BASIC_LOCATION = / and for proxypass configuration
```
location ~* /(ui|v1) {
...
}
```
I can open my url:elk-nomad-pro whith password. BUT:
1 I get error
` Sep 30 20:41:58 nginx: 2020/09/30 20:41:58 [error] 799#0: *1 lua entry thread aborted: runtime error: access_by_lua(dnsbl.conf:84):37: attempt to index local 'av' (a number value) stack traceback: coroutine 0: 	access_by_lua(dnsbl.conf:84): in main chunk, client: 3, server: elk-nomad-pro, request: "GET / HTTP/1.1", host: "elk-nomad-prod"`
then I commented row which inclued dnsbl.conf. Did reload of nginx and
On my URL is opened page with "Hello nginx!", and I have to write url:elk-nomad-prod/UI and only after this I able get access to my nomad web ui.

**Comment by bunkerity** [comment]
Hello @zakabluk, can I ask you to try again with the latest version ? Also, just set the string **sitewide** to the **AUTH_BASIC_LOCATION** environ variable : `docker run ... -e AUTH_BASIC_LOCATION=sitewide ...`

**Comment by zakabluk** [comment]
@bunkerity yes it helped me, but now I become get http response 403, always.

**Comment by bunkerity** [comment]
Can you check if your IP address is blacklisted in one of the following DNSBL : bl.blocklist.de problems.dnsbl.sorbs.net sbl.spamhaus.org xbl.spamhaus.org ?
Before we find a solution for this, you can set the environment variable `USE_DNSBL=no` instead of manually commenting the line.

**Comment by zakabluk** [comment]
@bunkerity ip is absent in spame bases. I want notice that in the past I got error

`Sep 30 20:41:58 nginx: 2020/09/30 20:41:58 [error] 799#0: *1 lua entry thread aborted: runtime error: access_by_lua(dnsbl.conf:84):37: attempt to index local 'av' (a number value) stack traceback: coroutine 0: access_by_lua(dnsbl.conf:84): in main chunk, client: 3, server: elk-nomad-pro, request: "GET / HTTP/1.1", host: "elk-nomad-prod"`

USE_DNSBL=no. - OK! Tnx.

**Comment by bunkerity** [comment]
Hello @zakabluk can you try again with the last version ? I've improved the logging and we should have more details in the error logs.

**Comment by zakabluk** [comment]
Hi @bunkerity ! I launched new container and nginx didn't start. In log is absent errors.
Only
```
[*] Updating clamav (in background) ...
[*] Running nginx ...
==> /var/log/access.log <==

==> /var/log/error.log <==
nginx: [alert] kill(916, 1) failed (3: No such process) 
```
Then I launched nginx is manual by command `nginx` and nginx was started without errors.

**Comment by bunkerity** [comment]
Looks like I can't reproduce this issue : 

```
$ docker pull bunkerity/bunkerized-nginx
$ mkdir www
$ echo "test" > www/test.html
$ docker run --rm -d -p 8000:80 -v "${PWD}/www:/www" --name "test" bunkerity/bunkerized-nginx
$ sleep 30
$ curl http://localhost:8000/test.html
test
```

Can you give more information about the environment variables, configs or your docker-compose.yml ?

**Comment by zakabluk** [comment]
HI! @bunkerity  i use nomad and bext variables:

```
       env {
        "SERVE_FILES" = "no"
        "USE_REQ_LIMIT" = "no"
        "USE_MODSECURITY" = "no"
        "USE_MODSECURITY_CRS" = "no"
        "AUTH_BASIC_LOCATION" = "sitewide"
        "REDIRECT_HTTP_TO_HTTPS" = "yes"
        "AUTO_LETS_ENCRYPT" = "yes"
        "DISABLE_DEFAULT_SERVER" = "yes"
        "USE_AUTH_BASIC" = "yes"
        "SERVER_NAME" = "XXXX XXXX XXXX XXXX"
        "AUTH_BASIC_USER" = "{{ auth_basic_user }}"
        "AUTH_BASIC_PASSWORD" =  "{{ auth_basic_password }}"
        "AUTH_BASIC_TEXT" = "This is a nomad WebUI. elk-nomad-pro - is default UI, elk-nomad - is modern, kibana is kibana "
        "USE_DNSBL" = "no"
       }
```

**Comment by bunkerity** [comment]
Hi @zakabluk, I still can't reproduce this issue at least with the last [1.1.1](https://github.com/bunkerity/bunkerized-nginx/releases/tag/v1.1.1) version. Can you update the image to the latest version and double check that you don't have any logs ?

**Comment by zakabluk** [comment]
@bunkerity hm, I deleted containers and cert. And now I launched container and get error
```
    raise errors.StandaloneBindError(error, port)
certbot.errors.StandaloneBindError: Problem binding to port 8080: Could not bind to IPv4 or IPv6.
```
from log
```
2020-11-03 21:42:23,238:DEBUG:certbot._internal.main:Arguments: ['--standalone', '-n', '--preferred-challenges', 'http', '-d', 'xxxx', '--email', 'contact@xxxx', '--agree-tos', '--http-01-po
rt', '8080']
```
what does it mean port 8080? I use 80/443 port for letsencrypt.
I set 
        "LISTEN_HTTP" = "80"
but it's not help.

**Comment by zakabluk** [comment]
@bunkerity  I disable https and just run container and got new error

```
2020-11-04 08:17:07,432 fail2ban.jail           [48516]: INFO    Jail 'nginx-filter' started
2020-11-04 08:17:07,449 fail2ban.utils          [48516]: #39-Lev. 7f4b622c3cb0 -- exec: echo "" > /etc/nginx/fail2ban-ip.conf && /usr/sbin/nginx -s reload
2020-11-04 08:17:07,449 fail2ban.utils          [48516]: ERROR   7f4b622c3cb0 -- stderr: 'nginx: [error] open() "/tmp/nginx.pid" failed (2: No such file or directory)'
2020-11-04 08:17:07,449 fail2ban.utils          [48516]: ERROR   7f4b622c3cb0 -- returned 1
2020-11-04 08:17:07,449 fail2ban.actions        [48516]: ERROR   Failed to start jail 'nginx-filter' action 'nginx-action': Error starting action Jail('nginx-filter')/nginx-action
```
```
        "SERVE_FILES" = "no"
#        "LISTEN_HTTP" = "80"
        "USE_REQ_LIMIT" = "no"
        "USE_MODSECURITY" = "no"
        "USE_MODSECURITY_CRS" = "no"
        "AUTH_BASIC_LOCATION" = "sitewide"
        "REDIRECT_HTTP_TO_HTTPS" = "no"
        "AUTO_LETS_ENCRYPT" = "no"
        "DISABLE_DEFAULT_SERVER" = "yes"
        "USE_AUTH_BASIC" = "yes"
#        "SERVER_NAME" = "xxxx.space"
        "AUTH_BASIC_USER" = "{{ auth_basic_user }}"
        "AUTH_BASIC_PASSWORD" =  "{{ auth_basic_password }}"
        "AUTH_BASIC_TEXT" = "This is a nomad WebUI. elk-nomad-pro - is default UI, elk-nomad - is modern, kibana is kibana "
        "USE_DNSBL" = "no"
```
and in server.conf was exist 
`listen 0.0.0.0:8080;`

**Comment by bunkerity** [comment]
The new version now listens on 8080 for HTTP and 8443 for HTTPS inside the container. You need to redirect ports like this : 

```shell
docker run ... -p 80:8080 -p 443:8443 ... bunkerity/bunkerized-nginx
```

You can find an example for reverse proxy [here](https://github.com/bunkerity/bunkerized-nginx/tree/master/examples/reverse-proxy).

**Comment by bunkerity** [comment]
I will add two environment variables to choose the HTTP and HTTPS ports.

**Comment by zakabluk** [comment]
> I will add two environment variables to choose the HTTP and HTTPS ports.

@bunkerity thanks, i will wait for you. Now I use version 1.0.0.

**Comment by bunkerity** [comment]
Hey @zakabluk you can now use the **HTTP_PORT** and **HTTPS_PORT** environment variables to set the ports of your choice. Can you try the [1.1.2](https://github.com/bunkerity/bunkerized-nginx/releases/tag/v1.1.2) and tell if it's ok for you ?

**Comment by bunkerity** [comment]
Closing this as it should now be fixed. Do not hesitate to open a new issue if needed.
